source/expressions.tex

```diff
@@ -2737,7 +2737,8 @@
 
 template<typename T> concept C = requires {
   typename T::inner;    // required nested member name
-  typename S<T>;        // required class template specialization
+  typename S<T>;        // required valid~(\ref{temp.names}) \grammarterm{template-id};
+                        // fails if substitution failure occurs for the second template parameter of \tcode{S}
```

Comment:
Thanks for the cross-reference.
"substitution failure" feels too unspecific here, because it could mean substitution failure while instantiating all of `S<T, X>` ("substitution for" is not really "substitution into"). What we mean is substitution failure into the template-head. Suggestion, a little more tailored to the example:
"fails if \tocde{T} has no nested type \tcode{type}"

Reply:
> "substitution failure" feels too unspecific here, because it could mean substitution failure while instantiating all of `S<T, X>` ("substitution for" is not really "substitution into").

Okay, I can change it to say "substitution into".

> What we mean is substitution failure into the template-head. Suggestion, a little more tailored to the example:
> "fails if \tocde{T} has no nested type \tcode{type}"

Well, it also fails if `0` cannot be implicitly converted to `T::type`.